akka-remote/src/main/scala/akka/remote/artery/Codecs.scala

```diff
@@ -364,6 +420,19 @@ private[remote] class Decoder(
         }
       }
 
+      private def applyIncomingInstruments(decoded: InboundEnvelope, headerBuilder: HeaderBuilder): Unit = {
```

Comment:
Sounds like a good suggestion. Would this end up after the call to `deserializeForArtery` in the `Deserializer`?